types/artifacts.d.ts

```diff
@@ -170,10 +168,20 @@ declare global {
 
       /** @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#Attributes */
       export interface LinkElement {
-        rel: string
-        href: string
+        /** The `rel` attribute of the link, normalized to lower case. @see https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types */
+        rel: 'alternate'|'canonical'|'dns-prefetch'|'preconnect'|'preload'|'stylesheet'|string;
+        /** The `href` attribute of the link or `null` if it was invalid in the header. */
+        href: string | null
+        /** The raw value of the `href` attribute. Only different from `href` when source is 'header' */
+        hrefRaw: string
+        /** The `hreflang` attribute of the link */
+        hreflang: string
+        /** The `as` attribute of the link */
         as: string
+        /** The `crossOrigin` attribute of the link */
         crossOrigin: 'anonymous'|'use-credentials'|null
+        /** Where the link was found, either in the DOM or in the headers of the main document */
```

Comment:
worth linking to https://html.spec.whatwg.org/multipage/links.html#body-ok (chart of which types of link elements are allowed in the body)?

(I find the ridiculously esoteric "body-ok" to be hilarious)

Reply:
Actually, should we filter (in a future PR) based on body-ok and not have them appear in the artifact if they aren't ok?

Only issue is that this chart isn't exhaustive and other specs are allowed to define body-okness for their own keywords.

Reply:
Ooooooooh *that's* what you were talking about. I assumed it was a typo lol 😆 Yeah I'll add the link.

I think we should probably leave them in the artifact so some point in the future we could have an audit that flags trying to use non `body-ok`, but flagging them